core/src/OpenMPTarget/Kokkos_OpenMPTarget_Exec.cpp

```diff
@@ -104,16 +104,19 @@ void OpenMPTargetExec::clear_scratch() {
 
 void* OpenMPTargetExec::get_scratch_ptr() { return m_scratch_ptr; }
 
-void OpenMPTargetExec::resize_scratch(int64_t reduce_bytes,
-                                      int64_t team_reduce_bytes,
-                                      int64_t team_shared_bytes,
-                                      int64_t thread_local_bytes) {
+void OpenMPTargetExec::resize_scratch(int64_t team_size, int64_t shmem_size_L0,
+                                      int64_t shmem_size_L1) {
   Kokkos::Experimental::OpenMPTargetSpace space;
-  int64_t total_size =
-      MAX_ACTIVE_TEAMS * reduce_bytes +         // Inter Team Reduction
-      MAX_ACTIVE_TEAMS * team_reduce_bytes +    // Intra Team Reduction
-      MAX_ACTIVE_TEAMS * team_shared_bytes +    // Team Local Scratch
-      MAX_ACTIVE_THREADS * thread_local_bytes;  // Thread Private Scratch
+  const int64_t shmem_size =
+      shmem_size_L0 + shmem_size_L1;  // L0 + L1 scratch memory per team.
+  const int64_t padding = shmem_size * 10 / 100;  // Padding per team.
```

Comment:
Over allocating the requested scratch memory by 10%  per team to find an open slot sooner.

Reply:
> ??? How is this supposed to work? Why `#if`?
> 
> https://github.com/kokkos/kokkos/blob/b3e6acd68c1874d59135a689704224fdf3f15ce1/core/src/impl/Kokkos_SharedAlloc.hpp#L51-L69

Yes, you are right. It has to be #ifdef here.